### ISSUE TITLE
`invalid byte sequence in UTF-8` error while linting regexp with `Lint/MixedCaseRange` cop

### ISSUE THREAD

**Issue by larouxn** [rubocop/rubocop#12703]
Running linting with the [Lint/MixedCaseRange](https://docs.rubocop.org/rubocop/cops_lint.html#lintmixedcaserange) cop enabled leads to an error when linting the following regular expression line. Surely a complex line but not a range in the first place but rather a regular expression. Plus, the line has been working fine in production for months so while I'm not 100% sure it's a RuboCop issue, I'm tempted to say it is.

```ruby
term.gsub(%r{[\s\.\-\–\&\,\:\;\!\?/\'\"\`\%\+\(\)\[\]]}, "").downcase
```

--------

## Expected behavior

No error nor linting issue is reported when running over the above line with the aforementioned cop.

## Actual behavior

```console
Inspecting 1 file
An error occurred while Lint/MixedCaseRange cop was inspecting some_file.rb:104:16.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Lint/MixedCaseRange cop was inspecting some_file.rb:104:16.
Errors are usually caused by RuboCop bugs.
```

Running with `--debug` flag, it appears the error is an invalid byte sequence. 🤔 

```console
An error occurred while Lint/MixedCaseRange cop was inspecting some_file.rb:104:16.
invalid byte sequence in UTF-8
```

## Steps to reproduce the problem

1. Add the `gsub` line above to a Ruby file.
2. Run `bundle exec rubocop --only Lint/MixedCaseRange --debug the_file.rb`.

## RuboCop version

```
❯ rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-factory_bot 2.25.1
  - rubocop-minitest 0.34.5
  - rubocop-performance 1.20.2
  - rubocop-rails 2.23.1
  - rubocop-sorbet 0.7.7
```

**Comment by larouxn** [comment]
Thanks a bunch for the fix! I can confirm that regex line lints fine now. 😌 

```ruby
Inspecting 1 file
.

1 file inspected, no offenses detected
```
